Explicitly mention kb/user callbacks as boolean (#18448)
qmk/qmk_firmware: docs/pr_checklist.md

    - for new MCUs, a new "child" keyboard should be added that targets your newly-added MCU, so that builds can be verified
    - for new hardware support such as display panels, core-side matrix implementations, or other peripherals, an associated keymap should be provided
    - if an existing keymap exists that can leverage this functionality this may not be required (e.g. a new RGB driver chip, supported by the `rgb` keymap) -- consult with the QMK Collaborators on Discord to determine if there is sufficient overlap already

> - any features adding kb/user callbacks must be boolean, to allow for user override of kb level callbacks. 

Reviewer: ```suggestion
- any features adding `_kb`/`_user` callbacks must return a `bool`, to allow for user override of keyboard-level callbacks. 
```
